src/utils/userLocation.ts

```diff
@@ -55,26 +86,32 @@ export const enterExternalRoom = ({ userId, room }: EnterExternalRoomProps) => {
 };
 
 export const useUpdateTimespentPeriodically = (
-  user: UserInfo | undefined,
-  roomName: string
+  roomName: string,
+  userId?: string
 ) => {
-  const shouldUseInterval = user && roomName;
+  const shouldUseInterval = userId && roomName;
 
   useInterval(
     () => {
-      // Time spent is currently counted multiple time if multiple tabs are open
-      if (!user || !roomName) return;
+      // @debt time spent is currently counted multiple time if multiple tabs are open
```

Comment:
[nit]
```suggestion
      // @debt time spent is currently counted multiple times if multiple tabs are open
```

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1129/commits/509bf9e36436cddbbfb4c24212ca2e48f29b2a6c